### DISCUSSION TITLE
Change PaymentMethod label based on cart value

### DISCUSSION THREAD

**Discussion by mihalikv** [woocommerce/woocommerce#50206] [Checkout Flow]
Hi,

I am working on a payment gateway plugin where you have just one option on the WooCommerce side, and as a user, you can select the method on the side of the payment provider if you want to use card/bank transfer/pay later.

I am trying to achieve different payment icons based on the cart total price. The use case is that when a user has a cart total above the limit (1000 euros), I want to show an additional "pay later" logo in my payment gateway.

However, the label component does not have access to the cart price in props; only the content component does.

Is it possible to change the payment label based on the cart price?

I've tried something like this, but it's not working because the content component is rendered only when the user has selected the payment.

```
const Content = (props) => {
    let description = decodeEntities(settings.description || '');
    let hidePayLaterLogo;
    let min_amount = settings.available_payment_methods?.PAY_LATER?.amount_range_rule?.min_amount || 0;
    let max_amount = settings.available_payment_methods?.PAY_LATER?.amount_range_rule?.max_amount || 0;

    min_amount *= 100;
    max_amount *= 100;
    if (min_amount <= props.billing.cartTotal.value && props.billing.cartTotal.value <= max_amount) {
        hidePayLaterLogo = <></>;
    } else {
        hidePayLaterLogo = <style>{`
        .wc-block-components-payment-method-icon--paylater {
            display: none !important;
        }       
    `}</style>
    }

    return (
        <>
            {hidePayLaterLogo}
            {description}
        </>
    )
}

const Label = (props) => {
    const {PaymentMethodIcons} = props.components
    return (
        <>
            {settings.icons.length > 0 && (
                <PaymentMethodIcons icons={settings.icons}/>
            )}
        </>
    );
}

registerPaymentMethod({
    name: payment_id,
    label: <Label/>,
    content: <Content/>,
    edit: <Content/>,
    canMakePayment: () => true,
    ariaLabel: label,
    supports: {
        showSaveOption: settings.showSaveOption ?? false,
        showSavedCards: settings.showSavedCards ?? false,
        features: settings.supports ?? ['products'],
    }
})
```

**Comment by senadir** [comment]
Hey! Sadly for now, the label icon is static and we're unlikely to pass it for now (as we're exploring future versions of the payment API).

As a workaround, you can hide it using CSS and JS subscribe.

Icons have a classname of `wc-block-components-payment-method-icon wc-block-components-payment-method-icon--${id}` assuming the one you're looking to hide is called `pay-later`, you can run the following psuedo code:

```js
import { subscribe, select } from '@wordpress/data';
import { CART_STORE_KEY } from '@woocommerce/block-data';

// Subscribe to the cart store.
let cachedCartTotals = null;
const unsubscribe = subscribe( () => {
	const { total_price: totals } = select( CART_STORE_KEY ).getCartTotals();
	if ( cachedCartTotals === totals ) {
		return;
	}

	cachedCartTotals = totals;

	const iconToHide = document.querySelector(
		'.wc-block-components-payment-method-icon--pay-later'
	);

	if ( iconToHide ) {
		if ( parseInt( cachedCartTotals, 10 ) < 10000 ) {
			iconToHide.style.display = 'none';
		} else {
			iconToHide.style.removeProperty('display');
		}
	}
	
}, CART_STORE_KEY );
```

Otherwise you can invert the logic:

- In your plugin, add CSS to hide the button by default.
- The code above would be checking the other way around.
- You would display it using it.
